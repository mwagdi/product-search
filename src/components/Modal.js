import React,{ Component } from 'react';
import Gallery from './Gallery';

class Modal extends Component{
    constructor(props) {
        super(props);
        this.modalRef = React.createRef();
    }
    componentDidMount(){
        document.addEventListener('click',this.handleClick);
    }
    componentWillUnmount(){
        document.removeEventListener('click',this.handleClick);
    }
    handleClick = e => {
        if(this.modalRef.current.contains(e.target)){
            return;
        }
        else{
            this.props.closeModal();
        }
    }
    render(){
        const { product,closeModal } = this.props;
        const images = product.additional_image_link.split(",");
        return (
            <div className="modal flex-container justify-center align-center">
                <div
                ref={this.modalRef}
                className="modal__wrap">
                    <div className="modal__header flex-container align-center">
                        <h3 className="modal__title flex-grow">{product.title}</h3>
                        <button
                        onClick={() => closeModal()}
                        className="modal__close">&times;</button>
                    </div>
                    <div className="modal__body">
                        <Gallery images={images} />
                    </div>
                </div>
            </div>
        )
    }
}

export default Modal;