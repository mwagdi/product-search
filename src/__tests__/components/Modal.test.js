import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Modal from '../../components/Modal';

describe("Modal component", () => {
    test('Renders Modal component', () => {
        const product = {
            title: "Product Title",
            additional_image_link: "*****,*****,*****"
        }
        const wrapper = shallow(<Modal product={product} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
})