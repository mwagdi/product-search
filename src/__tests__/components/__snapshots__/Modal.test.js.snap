// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Renders Modal component 1`] = `
<div
  className="modal flex-container justify-center align-center"
>
  <div
    className="modal__wrap"
  >
    <div
      className="modal__header flex-container align-center"
    >
      <h3
        className="modal__title flex-grow"
      >
        Product Title
      </h3>
      <button
        className="modal__close"
        onClick={[Function]}
      >
        Ã—
      </button>
    </div>
    <div
      className="modal__body"
    >
      <Gallery
        images={
          Array [
            "*****",
            "*****",
            "*****",
          ]
        }
      />
    </div>
  </div>
</div>
`;
